fenix/app/src/main/java/org/mozilla/fenix/compose/LinkText.kt

```diff
@@ -67,46 +67,21 @@ fun LinkText(
         linkTextDecoration,
     )
 
-    // When using UrlAnnotation, talkback shows links in a separate dialog and
-    // opens them in the default browser. Since this component allows the caller to define the
-    // onClick behaviour - e.g. to open the link in in-app custom tab, here StringAnnotation is used
-    // and modifier is enabled with Role.Button when screen reader is enabled.
```

Comment:
This was the reason why `StringAnnotation` was used. [PR](https://github.com/mozilla-mobile/firefox-android/pull/2370) for reference. As mentioned, with this change, talkback will open the links in a default browser. 

We have to make a choice:
1. Either we are fine with users being navigated out of firefox into their default browser and use `UrlAnnotation`.
2. Create a custom dialog that shows all the links so we can handle the `onClick` callbacks for those and open them in firefox.
3. Allow `LinkText` to support 2 behaviors (number 1 that's mentioned above and the changes in the PR) and let the callers decide, defaulting to 1, since Onboarding also uses it -> that means new users, who don't have firefox set as default. This will give us time to implement 2 properly. 

wdyt?

Reply:
I think at the very least we need to get Talkback working in general. Having only one link work with a11y services obviously isn't ideal, so as a first step, I'd like us to lean towards option 1 (for now).

Jeff and I talked about this a bit, and this is definitely a side-symptom of applying a desktop design convention to mobile of having 2+ clickable links in one body of text. iOS doesn't even comply with this. They are specced-out to have each of the privacy policy and terms of use links on their own lines.

![image](https://github.com/mozilla-mobile/firefox-android/assets/87384386/8ae9dbb9-616c-48a8-bc67-4bc58116251a)

So realistically, while we are trying to solve, in front of us, an engineering problem, we could definitely argue that it's instead a design + a11y problem. Having two links in one body of text makes more sense in a world where the user has the precision of a mouse pointer. What we have now is an a11y issue we're trying to code around, but is also an experience issue for everyone because the tap targets are way too small to have multi-embedded links. We kinda brush it under the rug here, likely because they are links to support information and not core parts of an experience.

Reply:
If I'm understanding that right, wouldn't it then be better to update the texts in both features (shopping, TPBM) to have one link and to keep using `StringAnnotation` and control the callback, since one link is already working on talkback. If we update to `UrlAnnotation` as this patch is doing, we potentially navigate users away firefox, definitely for onboarding, which would be a major bug IMO but I'm interested in opinions from UX/a11y on this.

Reply:
@MozillaNoah In these cases it is probably okay to push back (like the iOS team has) on the designs where they are deviating from the happy path. I agree with @rahulsainani that we should not be navigating users away from the app and we also should not be breaking Talkback.

Reply:
Forgot to loop back here last week, but I posted in the ENG-UX slack channel and the approach is being discussed amongst Design, Copy, and Localization because of various conflicts of those three channels/needs. The iOS design needs to be updated too as a result of that decision.